src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,197 @@
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
+import {
+  getCommentsByPostId,
+  deleteComment,
+} from '../../api/comments';
+import { NewCommentForm } from '../NewCommentForm';
+import { Loader } from '../Loader';
+import { ErrorNotification } from '../ErrorNotification';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+import { Error } from '../../types/Error';
+
+type Props = {
+  posts: Post[];
+  postId: number | null;
+  error: Error | null;
+  onError: (error: Error | null) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  posts,
+  postId,
+  error,
+  onError,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isCommentForm, setIsCommentForm] = useState(false);
+
+  const handleAddComment = useCallback((newComment) => {
+    setComments(allComments => [...allComments, newComment]);
+  }, []);
+
+  const currentPost
+    = useMemo(() => posts.filter(post => post.id === postId)[0], [postId]);
```

Comment:
Consider using Array.prototype.find instead of Array.prototype.filter to find a post by id, since filter will return an array and might introduce bugs if there are multiple posts with the same id, whereas find will return the first matching element or undefined if none are found.